catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RegistryPublicationServiceImpl.java

```
@@ -0,0 +1,211 @@
+        Metacard metacard = getMetacard(registryId);
+
+        Attribute locAttr = metacard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        if (locAttr == null || !locAttr.getValues()
+                .contains(destinationRegistryId)) {
+            return;
+        }
+
+        String sourceId = getSourceIdFromRegistryId(destinationRegistryId).orElse(null);
+
+        if (StringUtils.isNotBlank(sourceId)) {
+            federationAdminService.deleteRegistryEntriesByRegistryIds(Collections.singletonList(
+                    registryId), Collections.singleton(sourceId));
+
+            locAttr.getValues()
+                    .remove(destinationRegistryId);
+            metacard.setAttribute(locAttr);
+            metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.LAST_PUBLISHED,
+                    Date.from(ZonedDateTime.now()
+                            .toInstant())));
+
+            federationAdminService.updateRegistryEntry(metacard);
+        }
+    }
+
+    @Override
+    public void update(Metacard metacard) throws FederationAdminException {
+        Attribute publishedLocations =
+                metacard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        if (publishedLocations == null) {
+            return;
+        }
+
+        Set<String> locations = publishedLocations.getValues()
+                .stream()
+                .map(Object::toString)
+                .collect(Collectors.toCollection(HashSet::new));
+
+        if (CollectionUtils.isNotEmpty(locations)) {
+            try {
+                federationAdminService.updateRegistryEntry(metacard, locations);
+                metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.LAST_PUBLISHED,
+                        Date.from(ZonedDateTime.now()
+                                .toInstant())));
+                federationAdminService.updateRegistryEntry(metacard);
+            } catch (FederationAdminException e) {
```

I think we just want to throw the exception rather than catch it here